### DISCUSSION TITLE
IrSender.sendNEC doesn't work with Uno R4 Wifi

### DISCUSSION THREAD

**Discussion by DavidSimacek** [Arduino-IRremote/Arduino-IRremote#1156] [Help]
I use the library on the Uno R4. I took the latest commit as I noticed there is support now for the Uno R4.
The receiving part works absolutely fine. I run issue though when I try to transmit. I want to replicate my LG TV remote. So with the receiver I collected all the codes I needed and compared them with the reference online and all of them are as expected. 
Then I sent one of the code through the IrSender.sendNEC, but the TV didn't react. I tried to make a loopback setup and listen whan is sent (although I'm not really sure it is supposed to work correctly) and I get unknown protocol codes, every time something else even though I sent always the same (IrSender.sendNEC(0x4, 0x3, 0);)
I have the ability to look at the transmit path with my scope, but before I go this way, I wanted to check if in general the Uno R4 is already supported.
Thank you

**Comment by ArminJo** [comment]
Since I do not own a R4, I cannot support it well, but sending should work.
A Dump of the unknown protocol would be very helpful, to decide, what may be wrong.
Thank you

**Reply by DavidSimacek** [reply]
Armin, thanks for the support.
I used example Receive dump. I have the Uno R4, Tx on the pin 7, Rx on the pin 8. Then I send IrSender.sendNEC(0x4, 0x3, 0); and wait 15s before the same data send again. Here are two dumps. I decided for two as there is always something else received.
#1
17:39:27.368 ->  -3276750
17:39:27.368 ->  +9700,- 400
17:39:27.368 ->  + 550,- 450 + 500,- 500 + 600,- 400 + 500,- 500
17:39:27.410 ->  + 500,- 500 + 550,- 450 + 550,- 450 + 500,- 500
17:39:27.410 ->  + 550,- 450 + 500,- 500 + 600,- 400 + 500,- 500
17:39:27.410 ->  + 500,- 500 + 500,- 500 + 500,- 500 + 500,- 500
17:39:27.410 ->  + 500,- 500 + 500,- 500 + 500,- 500 + 550,- 450
17:39:27.410 ->  + 550,- 450 + 550,- 450 + 600,- 400 + 500,- 500
17:39:27.410 ->  + 500,- 500 + 600,- 400 + 500,- 500 + 500,- 500
17:39:27.410 ->  + 500,- 500 + 500,- 500 + 500,- 500 + 500
17:39:27.410 -> Sum: 41600
17:39:27.410 -> 
17:39:27.410 -> Result as internal 8bit ticks (50 us) array - compensated with MARK_EXCESS_MICROS=20
17:39:27.441 -> uint8_t rawTicks[65] = {194,8, 11,9, 10,10, 12,8, 10,10, 10,10, 11,9, 11,9, 10,10, 11,9, 10,10, 12,8, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 11,9, 11,9, 11,9, 12,8, 10,10, 10,10, 12,8, 10,10, 10,10, 10,10, 10,10, 10,10, 10};  // Protocol=UNKNOWN Hash=0x95DACC9 33 bits (incl. gap and start) received
17:39:27.441 -> 
17:39:27.441 -> Result as microseconds array - compensated with MARK_EXCESS_MICROS=20
17:39:27.483 -> uint16_t rawData[65] = {9680,420, 530,470, 480,520, 580,420, 480,520, 480,520, 530,470, 530,470, 480,520, 530,470, 480,520, 580,420, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 530,470, 530,470, 530,470, 580,420, 480,520, 480,520, 580,420, 480,520, 480,520, 480,520, 480,520, 480,520, 480};  // Protocol=UNKNOWN Hash=0x95DACC9 33 bits (incl. gap and start) received
17:39:27.483 -> 
17:39:27.483 -> Pronto Hex as string
17:39:27.483 -> char prontoData[] = "0000 006D 0021 0000 0176 000F 0016 0011 0014 0012 0018 000F 0014 0012 0014 0012 0016 0011 0016 0011 0014 0012 0016 0011 0014 0012 0018 000F 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0016 0011 0016 0011 0016 0011 0018 000F 0014 0012 0014 0012 0018 000F 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 06C3 ";

#2 
17:40:03.145 ->  -3276750
17:40:03.145 ->  +9000,-4450
17:40:03.145 ->  +1300,- 550 + 500,-1850 + 500,- 600 + 300,- 700
17:40:03.180 ->  + 350,- 650 + 600,- 550 + 550,- 650 + 550,-1750
17:40:03.180 ->  + 550,-1700 + 550,- 600 + 600,- 800 + 450,- 500
17:40:03.180 ->  + 500,- 500 + 500,- 500 + 500,- 500 + 500,- 500
17:40:03.180 ->  + 500,- 500 + 500,- 500 + 500,- 500 + 500,- 500
17:40:03.180 ->  + 500,- 500 + 500,- 500 + 500,- 500 + 500,- 500
17:40:03.180 ->  + 500,- 500 + 500,- 500 + 500,- 500 + 500,- 500
17:40:03.180 ->  + 600,- 400 + 500,- 500 + 550,- 450 + 600
17:40:03.212 -> Sum: 50750
17:40:03.212 -> 
17:40:03.212 -> Result as internal 8bit ticks (50 us) array - compensated with MARK_EXCESS_MICROS=20
17:40:03.212 -> uint8_t rawTicks[65] = {180,89, 26,11, 10,37, 10,12, 6,14, 7,13, 12,11, 11,13, 11,35, 11,34, 11,12, 12,16, 9,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 10,10, 12,8, 10,10, 11,9, 12};  // Protocol=UNKNOWN Hash=0x48F5BD2 33 bits (incl. gap and start) received
17:40:03.249 -> 
17:40:03.249 -> Result as microseconds array - compensated with MARK_EXCESS_MICROS=20
17:40:03.249 -> uint16_t rawData[65] = {8980,4470, 1280,570, 480,1870, 480,620, 280,720, 330,670, 580,570, 530,670, 530,1770, 530,1720, 530,620, 580,820, 430,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 480,520, 580,420, 480,520, 530,470, 580};  // Protocol=UNKNOWN Hash=0x48F5BD2 33 bits (incl. gap and start) received
17:40:03.285 -> 
17:40:03.285 -> Pronto Hex as string
17:40:03.285 -> char prontoData[] = "0000 006D 0021 0000 015B 00AA 0033 0014 0014 0046 0014 0016 000C 001A 000E 0018 0018 0014 0016 0018 0016 0043 0016 0041 0016 0016 0018 001E 0012 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0014 0012 0018 000F 0014 0012 0016 0011 0018 06C3 ";

Thank you.

**Reply by ArminJo** [reply]
Sorry I have no clue. 
There is obviously one (the first) bit missing, but this should not be R4 specific (or the WiFi part?).
Without a R4 at hand I am stuck.

**Reply by DavidSimacek** [reply]
@ArminJo , thank you for your help anyway. I actually work as an embedded SW guy, so I know how headache it is try to analyze something without being able to debug it on the target HW.
I will try to look myself. I have no previous experience with the IRRemote library and also Arduino in general, but at least now I have a reason to dive a little bit into it.
I will post here if I have some relevant findings regarding this topic.

**Reply by toseef-ahmd** [reply]
@DavidSimacek did you manage to solve your issue? Mind sharing here how you solved it?
Thanks
